src/Core/Commit.js

```diff
@@ -0,0 +1,363 @@
+	}).catch(function (e) {
+		errorBoundary(element, e, SharedSiteAsync+':'+SharedSiteRender, SharedErrorActive)
+	})
+}
+
+/**
+ * @param {Element} element
+ * @param {List} children
+ * @param {Element} host
+ * @param {number} signature
+ * @param {number} mode
+ */
+function commitChildren (element, children, host, signature, mode) {
+	var length = children.length
+	var sibling = children.next
+	var next = sibling
+
+	while (length-- > 0) {
+		if (next.DOM) {
+			sibling = next
+			children.insert(next = merge(new Element(SharedElementNode), next), sibling)
+			children.remove(sibling)
+		}
+
+		commitMount(next, element, element, host, signature, mode)
+		next = next.next
+	}
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} sibling
+ * @param {Element} parent
+ * @param {Element} host
+ * @param {number} signature
+ * @param {number} mode
+ */
+function commitMount (element, sibling, parent, host, signature, mode) {
+	element.host = host
+	element.parent = parent
+	element.context = host.context
+
+ 	switch (element.id) {
+ 		case SharedElementComponent:
+ 			element.work = SharedWorkTask
+ 			
+ 			commitMount(componentMount(element), sibling, parent, element, signature, mode)
+ 			element.DOM = element.children.DOM
+
+ 			if (element.ref)
+ 				commitReference(element, element.ref, SharedReferenceAssign)
+ 			
+ 			if (element.owner[SharedComponentDidMount])
+ 				getLifecycleMount(element, SharedComponentDidMount)
+
+ 			element.work = SharedWorkSync
+ 			return
+ 		case SharedElementPortal:
+ 			element.DOM = DOM(element.type)
+ 			break
+ 		case SharedElementPromise:
+ 			commitPromise(element, element)
+ 		case SharedElementFragment:
+ 			element.DOM = DOM(DOMTarget(parent))
+ 			break
+ 		case SharedElementNode:
+ 			element.xmlns = DOMType(element.type, parent.xmlns)
+ 		case SharedElementText:
+ 			switch (mode) {
+ 				case SharedMountClone:
+ 					if (element.DOM = DOMFind(element, parent))
+	 					break
+ 				default:
+ 					element.DOM = commitDOM(element)
+ 					
+ 					if (signature < SharedMountInsert)
+ 						commitAppend(element, parent)
+ 					else
+ 						commitInsert(element, sibling, parent)
+ 			}
+
+ 			if (element.id === SharedElementText)
+ 				return
+ 	}
+
+ 	commitChildren(element, element.children, host, SharedMountAppend, mode)
+ 	commitProperties(element, element.props, SharedPropsMount)
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} parent
+ * @param {number} signature
+ */
+function commitUnmount (element, parent, signature) {
+	if (element.id === SharedElementComponent)
+		return componentUnmount(element, element.children, parent, signature)
+
+	commitRemove(element, parent)
+	commitDetach(element, signature)
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} snapshot
+ * @param {number} signature
+ */
+function commitReplace (element, snapshot, signature) {
+	if (signature === SharedMountReplace && commitUnmount(element, element.parent, SharedMountReplace))
```

Comment:
this branch is never taken because `commitUnmount` never returns a value because `componentUnmount` doesn't exist. Technically a problem with `commitUnmount`, but tests should have triggered this branch too

Reply:
`componentUnmount` does exist, [src/Core/Component.js#L152-L161#](https://github.com/thysultan/dio.js/blob/V8/src/Core/Component.js#L152-L161)

Reply:
thanks, github had suppressed the diff